sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/AutoCheckpointer.java

```diff
@@ -58,10 +65,15 @@ public void close(ChangeFeedObserverContext context, ChangeFeedObserverCloseReas
 
         if (this.isCheckpointNeeded()) {
             return context.checkpoint()
-                .doOnSuccess((Void) -> {
+                .onErrorResume(throwable -> {
```

Comment:
`onErrorResume` is for when you are changing the return value, which is not the case here.
`doOnError` is more suitable for these cases
https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#doOnError-java.util.function.Consumer-

Reply:
Fixed